src/asyncProfiler.cpp

```diff
@@ -145,13 +146,35 @@ void Profiler::recordSample(void* ucontext) {
 
     ASGCT_CallFrame frames[MAX_FRAMES];
     ASGCT_CallTrace trace = {jni, MAX_FRAMES, frames};
-    AsyncGetCallTrace(&trace, trace.num_frames, ucontext);
+    if (asgct == NULL) {
+        const char ERROR[] = "No AsyncGetCallTrace";
+        write(STDOUT_FILENO, ERROR, sizeof (ERROR));
+        return;
+    }
+
+    (*asgct)(&trace, trace.num_frames, ucontext);
 
     if (trace.num_frames > 0) {
         storeCallTrace(&trace);
         storeMethod(frames[0].method_id);
+    } else if (trace.num_frames == 0) {
```

Comment:
These constants are copied from http://hg.openjdk.java.net/jdk8/jdk8/hotspot/file/tip/src/share/vm/prims/forte.cpp and are not used anywhere else:
```c++
// These name match the names reported by the forte quality kit
enum {
  ticks_no_Java_frame         =  0,
  ticks_no_class_load         = -1,
  ticks_GC_active             = -2,
  ticks_unknown_not_Java      = -3,
  ticks_not_walkable_not_Java = -4,
  ticks_unknown_Java          = -5,
  ticks_not_walkable_Java     = -6,
  ticks_unknown_state         = -7,
  ticks_thread_exit           = -8,
  ticks_deopt                 = -9,
  ticks_safepoint             = -10
};
```
Should I just put a reference to `forte.cpp` in a preceding comment?